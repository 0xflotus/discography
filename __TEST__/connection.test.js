import testSnapshot from './fixtures/testSnapshot';

test('Should return correct pageInfo when source edge is null', async () => {
  const query = `
    query {
      search {
        artist (name: "DSIKFOIJSDOIFJOISAJDOIJ") {
          edges {
            node {
              id
            }
            cursor
          }
          pageInfo {
            hasNextPage
            endCursor
          }
          totalReturned
        }
      }
    }
  `;

  await testSnapshot(query);
});

test('Should apply correct pagination when cursor is provided', async () => {
  const query = `
    query {
      search {
        release (title: "hyph mngo", first: 1, after: "MTk2Njc3OQ==") {
          edges {
            node {
              id
            }
            cursor
          }
          pageInfo {
            hasNextPage
            endCursor
          }
          totalReturned
        }
      }
    }
  `;

  await testSnapshot(query);
});
