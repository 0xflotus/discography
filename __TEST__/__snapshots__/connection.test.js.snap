// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should apply correct pagination when cursor is provided 1`] = `
Object {
  "data": Object {
    "search": Object {
      "release": Object {
        "edges": Array [
          Object {
            "cursor": "MjA1NjUzMg==",
            "node": Object {
              "id": 2056532,
            },
          },
        ],
        "pageInfo": Object {
          "endCursor": "MjA1NjUzMg==",
          "hasNextPage": false,
        },
        "totalReturned": 1,
      },
    },
  },
}
`;

exports[`Should return correct pageInfo when source edge is null 1`] = `
Object {
  "data": Object {
    "search": Object {
      "artist": Object {
        "edges": Array [],
        "pageInfo": Object {
          "endCursor": null,
          "hasNextPage": null,
        },
        "totalReturned": 0,
      },
    },
  },
}
`;
